From Robert.Lukassen@tomtom.com  Thu Apr 29 12:38:47 2010
From: "Robert Lukassen" <Robert.Lukassen@tomtom.com>
Date: Tue, 30 Mar 2010 14:14:01 +0200
Subject: usb: gadget: Allow larger configuration descriptors
Message-ID: <D6DB9C7EDECDA944B870F62B5870086229B68C@NL-EXC-06.intra.local>


The composite framework allows gadgets with more than one function. This
can lead to situations where the configuration descriptor is larger than
the maximum of 512 bytes currently allowed by the composite framework.
This patch proposes to double that limit to 1024.

Signed-off-by: Robert Lukassen <robert.lukassen@tomtom.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/gadget/composite.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/gadget/composite.c
+++ b/drivers/usb/gadget/composite.c
@@ -36,7 +36,7 @@
  */
 
 /* big enough to hold our biggest descriptor */
-#define USB_BUFSIZ	512
+#define USB_BUFSIZ	1024
 
 static struct usb_composite_driver *composite;
 
